### ISSUE TITLE
Different camera behaviours issue.

### ISSUE THREAD

**Issue by Mannbaerschwein** [Nibre/MotherVR#4]
@Nibre, you probably know this already, but I just discovered another weird thing about the vr implementation.

So basicially, the player camera has 2 different states, which are switching constantly while playing.-
one is better for VR, the other one gives one a worse experience.

1. has headbobbing to the playermovement and causes a little off-centered headrotation.
2. has no headbobbing and headtracking (rotation) is solid.
 
switching between the 2 states:
1 is active while crouching and continues to stay active after standing up again....until:
2 gets active when aiming with a weapon, using the motion tracker, or when sprinting.

Do you think there is a way to have state 2 always on? or in other words, to get rid of the headbobbing and off.center issue at all times?

**Comment by jonathancumming** [comment]
You want to completely get rid of head bobbing imho...
It's one big thing that made people sick in Doom3 VR until it was removed...

-Just want to say, great work so far mate. Love it!

**Comment by Nibre** [comment]
That was something I noticed as well, but couldn't figure out how to recreate consistently, so having more information on what does and does not cause it is helpful. Hopefully with that info I can get it to stay that way state, should help a ton with comfort

**Comment by MulleredOG** [comment]
I think the off centered head rotation is linked to the movement of the character, this is noticeable when looking at your (virtual) feet and turning on the spot.  The movement is circular rather than pivotal

**Comment by MulleredOG** [comment]
I've just noticed something which might help find a fix for this issue - In survival mode, at the end after you've pressed the elevator button you get taken to a stats screen, in the stats screen you can still move around and look at your surroundings but the head bobbing and off centred positional tracking is not present, the tracking is perfect and centred like it should be

**Comment by Nibre** [comment]
Yes, that's what I've observed as well. You get into the 'gliding' state after you; Interact with most objects, or start/stop sprinting. Crouching, and some other things get you back to the 'bobbing' state. This is why the beginning of the game can be so sickening, as you can't sprint for a while.

**Comment by Mannbaerschwein** [comment]
There is another camera issue in the final section of the game, where everything is shaking and the headtracking orientation is messed up.
As far as I remember this was also the case on the DK2.
Not sure if this can be fixed, but it would save many poeple from feeling sick in that section.

**Comment by Nibre** [comment]
Alright, I've been able to force 'gliding' anytime you're walking/running, which is good. Haven't been able to code it up nicely yet, but it's working as a proof of concept. Crouching still seems to be a problem though, not sure if it's possible to remove it there yet. Other than that, it basically fixes this issue. 

The off center rotation thing is present in both states, it has to do with how close you are to your 're-center' point, or your 'body' basically. If you lean forward like 4 inches, rotation will be fine.

**Comment by Nibre** [comment]
I think I may be able to fix crouch 'bobbing' now too, fingers crossed. 

For the part at the end @Mannbaerschwein , can you link me to a YouTube video or something that shows that part? While I've seen the whole game, I do 95% of my quick testing at one good testing spot, so I'll need some more info to test the part you're talking about

**Comment by Mannbaerschwein** [comment]
That sounds great about the bobbing!

This is the part I meant: https://youtu.be/1pvaw3tqgaw?t=28953 (8:02.00)
The tracking orientation issue starts from there on and lasts until the final spacewalk scene.
Also, there is constant camera shaking, which however is not as bad as the orientation thing I'd say.
